src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_blocked_opt.cpp

```diff
@@ -0,0 +1,378 @@
+    DispatchData dispatchData;
+    auto in_layout = params.inputs[0].GetLayout();
+    auto out_layout = params.outputs[0].GetLayout();
+    std::vector<std::vector<Tensor::DataChannelName>> dims_by_gws = {{Tensor::DataChannelName::FEATURE},
+                                                                     {Tensor::DataChannelName::X, Tensor::DataChannelName::Y},
+                                                                     {Tensor::DataChannelName::BATCH}};
+    // Global workgroup size 0: feature, 1: spatial, 2: batch
+    dispatchData.gws[0] = CeilDiv(params.outputs[0].Feature().v, GetFeatureBlockSizeFromFormat(params, 0));
+    dispatchData.gws[2] = params.outputs[0].Batch().v;
+    if (DataTensor::ChannelsCount(params.outputs[0].GetLayout()) == 5)
+        dispatchData.gws[1] = params.outputs[0].X().v * params.outputs[0].Y().v * params.outputs[0].Z().v;
+    else if (DataTensor::ChannelsCount(params.outputs[0].GetLayout()) == 4)
+        dispatchData.gws[1] = params.outputs[0].X().v * params.outputs[0].Y().v;
+    else
+        IE_ASSERT("Unexpected dimension for eltwise_blocked_opt kernel.");
+
+    // Calculate local workgroup size
+    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo, in_layout, out_layout, dims_by_gws);
+
+    return dispatchData;
+}
+
+// Local
+static inline int GetFeatureBlockSizeFromFormat(const eltwise_params& arg, size_t index) {
+    auto in_layout = arg.inputs[index].GetLayout();
+    switch (in_layout) {
+    case DataLayout::b_fs_yx_fsv4:
+        return 4;
+    case DataLayout::b_fs_yx_fsv16:
+    case DataLayout::b_fs_yx_fsv32:
+    case DataLayout::b_fs_zyx_fsv16:
+    case DataLayout::b_fs_zyx_fsv32:
+    case DataLayout::bs_fs_yx_bsv32_fsv32:
+    case DataLayout::bs_fs_yx_bsv32_fsv16:
+    case DataLayout::bs_fs_yx_bsv16_fsv32:
+    case DataLayout::bs_fs_yx_bsv16_fsv16:
+    case DataLayout::bs_fs_zyx_bsv32_fsv32:
+    case DataLayout::bs_fs_zyx_bsv32_fsv16:
+    case DataLayout::bs_fs_zyx_bsv16_fsv32:
+    case DataLayout::bs_fs_zyx_bsv16_fsv16:
+        return 8;
+    default:
+        return 1;
+    }
+}
+
+static inline bool IsBroadcastingPossibleInput(const DataTensor& input, const DataTensor& output) {
+    if ((input.LogicalSize() == 1) ||
+        (input.LogicalSize() == output.Feature().v && input.Feature().v == output.Feature().v)) {
+            return true;
+        }
+    return false;
+}
+
+static inline bool InputHasFeatureBroadcast(const eltwise_params& params, const size_t op_num, const size_t input_idx) {
+    const auto &ew = params.operations[op_num];
+
+    const auto &input = ew.inputs[input_idx];
+    if (input.mode == EltwiseInputMode::INPUT_BUFFER) {
+        if (params.inputs[input_idx].LogicalSize() != 1
+            && params.inputs[input_idx].Feature().v == 1
+            && params.outputs[0].Feature().v != 1) {
+                return true;
+            }
+    }
```

Comment:
this is basically same as `IsBroadCastingPossibleInput`. What about reusing the function?

Reply:
InputHasFeatureBroadcast and IsBroadCastingPossibleInput have a little bit different condition to return true. It needs additional condition to use IsBroadCastingPossibleInput here. Here the condition only finds an input which does not have vector size. But IsBroadCastingPossibleInput is different.
 So I think we do not need to reuse the code.